safekeeper/src/http/routes.rs

```diff
@@ -373,8 +378,53 @@ async fn dump_debug_handler(mut request: Request<Body>) -> Result<Response<Body>
         .await
         .map_err(ApiError::InternalServerError)?;
 
-    // TODO: use streaming response
-    json_response(StatusCode::OK, resp)
+    let started_at = std::time::Instant::now();
+
+    let (tx, rx) = mpsc::channel(1);
+
+    let body = Body::wrap_stream(ReceiverStream::new(rx));
+
+    let mut writer = ChannelWriter::new(128 * 1024, tx);
+
+    let response = Response::builder()
+        .status(200)
+        .header(hyper::header::CONTENT_TYPE, "octet-stream")
+        .body(body)
+        .unwrap();
+
+    let span = info_span!("blocking");
+    tokio::task::spawn_blocking(move || {
+        let _span = span.entered();
+
+        let mut bytes: Vec<u8> = Vec::new();
+        serde_json::to_writer(&mut bytes, &resp).expect("failed to convert to bytes");
```

Comment:
We can't serialize everything to a Vec, it will store all serialized data in RAM. It should be possible to write directly to `writer` here.

Reply:
Oh, okay.
Seems that I missed the point.